src/main/java/slt/connectivity/google/GoogleClient.java

```diff
@@ -0,0 +1,155 @@
+    private static final String ERROR_MESSAGE = "Fout bij versturen.";
+
+    @Autowired
+    RestTemplate restTemplate;
+
+    @Autowired
+    GoogleConfig googleConfig;
+
+    public void sendMail(Oath2Token oath2Token, Message message) throws IOException, GeneralSecurityException {
+
+        if (oath2Token == null){
+            log.error("Unable to send mail. No token available");
+            return;
+        }
+
+        GoogleCredential credential = new GoogleCredential().setAccessToken(oath2Token.getAccess_token());
+
+        HttpTransport httpTransport = GoogleNetHttpTransport.newTrustedTransport();
+        JsonFactory jsonFactory = JacksonFactory.getDefaultInstance();
+        try {
+            com.google.api.services.gmail.model.Message googleMessage = createMessageWithEmail(message);
+            final Gmail service = new Gmail.Builder(httpTransport, jsonFactory, credential)
+                    .setApplicationName( googleConfig.getApplicationName())
+                    .build();
+            final com.google.api.services.gmail.model.Message execute = service
+                    .users()
+                    .messages()
+                    .send("me", googleMessage)
+                    .execute();
+            log.debug("Mail {} send.",execute.getId());
+        } catch (MessagingException e) {
+            log.error("Error during sending mail",e);
+        }
+    }
+
+    public MimeMessage createEmail(String to,
+                                          String from,
+                                          String subject,
+                                          String bodyText)
+            throws MessagingException {
+        Properties props = new Properties();
+        Session session = Session.getDefaultInstance(props, null);
+
+        MimeMessage email = new MimeMessage(session); // NOSONAR
+
+        email.setFrom(new InternetAddress(from));
+        email.addRecipient(javax.mail.Message.RecipientType.TO,
+                new InternetAddress(to));
+        email.setSubject(subject);
+        email.setContent(bodyText, "text/html");
+        return email;
+    }
+
+    com.google.api.services.gmail.model.Message createMessageWithEmail(Message emailContent)
+            throws MessagingException, IOException {
+        ByteArrayOutputStream buffer = new ByteArrayOutputStream();
+        emailContent.writeTo(buffer);
+        byte[] bytes = buffer.toByteArray();
+        String encodedEmail = Base64.encodeBase64URLSafeString(bytes);
+        com.google.api.services.gmail.model.Message message = new com.google.api.services.gmail.model.Message();
+        message.setRaw(encodedEmail);
+        return message;
+    }
+
+    public Oath2Token getAuthorizationToken(String authorizationCode)  {
+
+        String clientId = googleConfig.getClientId();
+        String clientSecret = googleConfig.getClientSecret();
+
+        Map reqPayload = new HashMap();
```

Comment:
Generics "Map<string, string> blabla = new HashMap<>()

Reply:
aangepast (ook in strava client)